src/ProjectTemplates/test/Templates.Tests/NewlineEndingTest.cs

```diff
@@ -0,0 +1,91 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.AspNetCore.InternalTesting;
+using Templates.Test.Helpers;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Templates.Test;
+
+public class NewlineEndingTest : LoggedTest
+{
+    private ITestOutputHelper _output;
+    public ITestOutputHelper Output
+    {
+        get
+        {
+            if (_output == null)
+            {
+                _output = new TestOutputLogger(Logger);
+            }
+            return _output;
+        }
+    }
+
+    [Theory]
+    [InlineData("Web.ProjectTemplates")]
+    [InlineData("Web.ItemTemplates")]
+    [InlineData("Web.Client.ItemTemplates")]
+    public void TemplateFiles_ShouldEndWithNewline(string projectName)
+    {
+        var templateDirectoryPath = GetTemplateDirectoryPath(projectName);
+
+        var filesWithoutNewlineEnding = new List<string>();
+
+        // Get all template source files (excluding third-party libraries and auto-generated localization files)
+        var files = Directory.GetFiles(templateDirectoryPath, "*.cs", SearchOption.AllDirectories)
+            .Concat(Directory.GetFiles(templateDirectoryPath, "*.fs", SearchOption.AllDirectories))
+            .Concat(Directory.GetFiles(templateDirectoryPath, "*.razor", SearchOption.AllDirectories))
+            .Concat(Directory.GetFiles(templateDirectoryPath, "*.cshtml", SearchOption.AllDirectories))
+            .Concat(Directory.GetFiles(templateDirectoryPath, "*.css", SearchOption.AllDirectories))
+            .Concat(Directory.GetFiles(templateDirectoryPath, "*.js", SearchOption.AllDirectories))
+            .Concat(Directory.GetFiles(templateDirectoryPath, "*.ts", SearchOption.AllDirectories))
+            .Concat(Directory.GetFiles(templateDirectoryPath, "*.tsx", SearchOption.AllDirectories))
+            .Concat(Directory.GetFiles(templateDirectoryPath, "*.html", SearchOption.AllDirectories))
+            .Concat(Directory.GetFiles(templateDirectoryPath, "*.json", SearchOption.AllDirectories))
+            .Concat(Directory.GetFiles(templateDirectoryPath, "*.xml", SearchOption.AllDirectories))
+            .Concat(Directory.GetFiles(templateDirectoryPath, "*.csproj", SearchOption.AllDirectories))
+            .Concat(Directory.GetFiles(templateDirectoryPath, "*.fsproj", SearchOption.AllDirectories))
+            .Where(f => !f.Contains($"{Path.DirectorySeparatorChar}wwwroot{Path.DirectorySeparatorChar}lib{Path.DirectorySeparatorChar}")) // Exclude third-party libraries
+            .Where(f => !f.Contains($"{Path.DirectorySeparatorChar}.template.config{Path.DirectorySeparatorChar}localize{Path.DirectorySeparatorChar}")); // Exclude auto-generated localization files in localize directory
```

Comment:
It'd be nice if we just fix these as well. Many editors want to add the newline when touching them locally and fighting against that is painful.